### PR TITLE
Upgrade TensorRT to 8.5.3

### PR DESCRIPTION
**Pull request by NateMeyer**
This pull request updates the TensorRT libraries to 8.5.3 in order to support the latest GPUs from Nvidia. Should address Issue https://github.com/blakeblackshear/frigate/issues/6666

This update also is able to do the model conversion from within the frigate image, which allows us to integrate it into the startup scripts.

Tasks:

- [x]  Update Libraries
- [x]  Update scripts to convert models locally
- [x]  Setup scripts to convert models automatically
- [x]  Update documentation
- [ ]  Testing

### CODE REVIEW

**Comment by NateMeyer** [comment]
I have a test image pushed if anyone wants to help verify this works on new (4000 series) and old (900 series) GPUs.  This change also streamlines the model generation process, and no pre-conversion steps are necessary, just add the model you want to use as an environment variable in your `docker-compose.yml`.

Using image `ghcr.io/natemeyer/frigate:trt-8.5-7761a1a-tensorrt`, add the following to your docker-compose file:

```yml
services:
  frigate:
    image: ghcr.io/natemeyer/frigate:trt-8.5-7761a1a-tensorrt
    environment:
      - USE_FP16=false
      - YOLO_MODELS=yolov7-tiny-416
```

The trt-models folder no longer needs to be mapped as a volume into the container.  The model in the `YOLO_MODELS` variable will be downloaded, converted, and saved into the path `/config/model_cache/tensorrt` when the container starts up.

You will also need to add this new path into the `config.yml` to point to the converted model.

```yml
model:
  path: /config/model_cache/tensorrt/yolov7-tiny-416.trt
```

Note: I had issues using the latest driver on my 1080ti, the system would crash when running object detection.  I rolled back to the 531.68 driver and it is working fine.  Not sure if this issue extends to any other cards or setups, or if I'm the lucky one.

**Comment by doctorttt** [comment]
Hi Nate. I followed your instructions above and ```ghcr.io/natemeyer/frigate:trt-8.5-7761a1a-tensorrt``` works on my 4060ti. The whole installation process was smooth and flawless. Please see the screenshots pasted here for your reference.  By the way, are the other yolo7 models available for testing: ```yolov7x-320, yolov7x-640, yolov7-640, yolov7-320```.  Also, are yolo8 models available for testing? Thanks

image

image

image

```
detectors:
  tensorrt:
    type: tensorrt
    device: 0 

model:
  path: /config/model_cache/tensorrt/yolov7-tiny-416.trt 
  input_tensor: nchw
  input_pixel_format: rgb
  width: 416
  height: 416
```

**Comment by NateMeyer** [comment]
Yes, all the same models should be available as before.  Just set them in the `YOLO_MODELS` variable and point to the model in the config file.  You can use a comma-separated list to convert multiple models just like the previous process.  The yolo example repo from yeahme49 didn't include any yolov8 models to my knowledge.

**Comment by doctorttt** [comment]
> Yes, all the same models should be available as before. Just set them in the `YOLO_MODELS` variable and point to the model in the config file. You can use a comma-separated list to convert multiple models just like the previous process. The yolo example repo from yeahme49 didn't include any yolov8 models to my knowledge.

I tried the other yolo7 models but couldn't get them to work. I'm stuck with ```unable to start service trt-model-prepare``` error message. Please advise. Thanks Nate

image

**Comment by NateMeyer** [comment]
@doctorttt Thanks for the heads up.  I think the conversion script was too verbose, so I redirected the stdout to /dev/null.  Give it a try with the latest update.  Pushing a new image, `ghcr.io/natemeyer/frigate:trt-8.5-3b21f86-tensorrt`.  If you see this issue again, I'll need more of the log to see what it's complaining about.

**Comment by doctorttt** [comment]
> ghcr.io/natemeyer/frigate:trt-8.5-3b21f86-tensorrt

@NateMeyer ```ghcr.io/natemeyer/frigate:trt-8.5-3b21f86-tensorrt``` works! I was able to get ```yolov7x-320, yolov7x-640, yolov7-640, yolov7-320``` to work on my 4060ti. The conversion for each went smoothly and no complaints in frigate.

**Review by NickM-27 — APPROVED** [review]

**Comment by NateMeyer** [comment]
@doctorttt One last note, you should be able to set the `USE_FP16` flag to true.  I believe that is primarily for older generation cards that don't have FP16 support like Pascal and Maxwell.  If you change that and want to regenerate the models, just clear out that model_cache folder.

File: docker/support/tensorrt_detector/tensorrt_libyolo.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+set -euxo pipefail
+
+SCRIPT_DIR="/usr/local/src/tensorrt_demos"
+
+# Clone tensorrt_demos repo
+git clone --depth 1 https://github.com/NateMeyer/tensorrt_demos.git -b conditional_download
+
+# Build libyolo
+cd ./tensorrt_demos/plugins && make all
```
Comment (madsciencetist) [inline docker/support/tensorrt_detector/tensorrt_libyolo.sh:11]: @NateMeyer Does this work in CI? The Makefile tries to read your device's CC

Comment (madsciencetist) [reply docker/support/tensorrt_detector/tensorrt_libyolo.sh:11]: https://github.com/blakeblackshear/frigate/actions/runs/5479330030/jobs/9981018663#step:11:703

**Comment by doctorttt** [comment]
> @doctorttt One last note, you should be able to set the `USE_FP16` flag to true. I believe that is primarily for older generation cards that don't have FP16 support like Pascal and Maxwell. If you change that and want to regenerate the models, just clear out that model_cache folder.

@NateMeyer Thanks - I flipped ```USE_FP16``` to true and regenerated all yolo7 models - no issue. :)

File: docker/support/tensorrt_detector/tensorrt_libyolo.sh (hunk 1 above)
Comment (madsciencetist) [reply docker/support/tensorrt_detector/tensorrt_libyolo.sh:11]: Ah, it still compiles, but defaults to the greatest-compatibility `-arch` compute. That's probably fine. Ok.

**Comment by madsciencetist** [comment]
@trixor you'll need to do the conversion yourself and supply the `.trt` file. However, it would be straightforward to upgrade `docker/support/tensorrt_detector/rootfs/etc/s6-overlay/s6-rc.d/trt-model-prepare/run` to be able to ingest a user-provided YOLO model.

**Pull request merged**